core/domain/classroom_config_domain.py

```diff
@@ -152,46 +257,100 @@ def require_valid_url_fragment(cls, url_fragment: str) -> None:
 
         if url_fragment == '':
             raise utils.ValidationError(
-                'Url fragment field should not be empty')
+                'Url fragment field should not be empty'
+            )
 
         utils.require_valid_url_fragment(
             url_fragment, 'Classroom URL Fragment',
-            constants.MAX_CHARS_IN_CLASSROOM_URL_FRAGMENT)
+            constants.MAX_CHARS_IN_CLASSROOM_URL_FRAGMENT
+        )
 
-    def validate(self) -> None:
-        """Validates various properties of the Classroom."""
+    @classmethod
+    def require_valid_thumbnail(
+        cls, thumbnail_filename: str, thumbnail_bg_color: str
+                            ) -> None:
+        """Check classroom thumbnail filename and background color validity.
 
-        if not isinstance(self.classroom_id, str):
+        Args:
+            thumbnail_filename: str. The thumbnail filename to validate.
+            thumbnail_bg_color: str. The bg color to validate.
+        """
+        if not isinstance(thumbnail_bg_color, str):
             raise utils.ValidationError(
-                'Expected ID of the classroom to be a string, received: %s.'
-                % self.classroom_id)
+                'Expected thumbnail_bg_color of the classroom to be a string, '
+                'received: %s.' % thumbnail_bg_color)
+        if thumbnail_filename == '':
+            raise utils.ValidationError(
+                'thumbnail_filename field should not be empty')
 
-        self.require_valid_name(self.name)
+        utils.require_valid_thumbnail_filename(thumbnail_filename)
 
-        self.require_valid_url_fragment(self.url_fragment)
+        if thumbnail_bg_color == '':
+            raise utils.ValidationError(
+                'thumbnail_bg_color field should not be empty')
 
-        if not isinstance(self.course_details, str):
+        if (
+            thumbnail_bg_color not in (
+                constants.ALLOWED_THUMBNAIL_BG_COLORS['classroom']
+            )):
             raise utils.ValidationError(
-                'Expected course_details of the classroom to be a string, '
-                'received: %s.' % self.course_details)
+                'Classroom thumbnail background color %s is '
+                'not supported.' % thumbnail_bg_color
+            )
+
+    @classmethod
+    def require_valid_banner(
```

Comment:
This function and the function above are the same, right? Let's extract common functionality into one function

Reply:
bg color is common so made a function for it only and we have a utility function to test for filename for thumbnail and general image so used it.